### ISSUE TITLE
ACL is not appear

### ISSUE THREAD

**Issue by mikhailznak** [confluentinc/terraform-provider-confluentcloud#38]
Terraform: `v1.1.2`
confluentcloud: `0.3.0`

```
resource "confluentcloud_kafka_acl" "test" {
  provider      = confluentcloud
  kafka_cluster = module.kafka.cluster
  resource_type = "TOPIC"
  resource_name = "test"
  pattern_type  = "LITERAL"
  principal     = "User:${module.kafka.service_account}"
  operation     = "WRITE"
  permission    = "ALLOW"
  http_endpoint = module.kafka.http_endpoint
  credentials {
    key    = data.google_secret_manager_secret_version.key.secret_data
    secret = data.google_secret_manager_secret_version.secret.secret_data
  }
}
```

Terraform applies current configuration but acl is not appear.
Topic and user exist

Terraform Debug log 

```
confluentcloud_kafka_acl.test: Creating...
2022-01-27T18:28:35.591+0300 [INFO]  Starting apply for confluentcloud_kafka_acl.test
2022-01-27T18:28:35.592+0300 [DEBUG] confluentcloud_kafka_acl.test: applying the planned Create change
2022-01-27T18:28:35.592+0300 [TRACE] GRPCProvider: ApplyResourceChange
2022-01-27T18:28:36.382+0300 [INFO]  provider.terraform-provider-confluentcloud_0.3.0: 2022/01/27 18:28:36 [DEBUG] Created kafka ACL lkc-eee34/TOPIC#test#LITERAL#User:sa-23ff3t#*#WRITE#ALLOW: timestamp=2022-01-27T18:28:36.382+0300
2022-01-27T18:28:36.383+0300 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for confluentcloud_kafka_acl.test
2022-01-27T18:28:36.383+0300 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: writing state object for confluentcloud_kafka_acl.test
2022-01-27T18:28:36.383+0300 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for confluentcloud_kafka_acl.test
2022-01-27T18:28:36.383+0300 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: writing state object for confluentcloud_kafka_acl.test
confluentcloud_kafka_acl.test: Creation complete after 0s [id=lkc-eee34/TOPIC#test#LITERAL#User:sa-23ff3t#*#WRITE#ALLOW]
2022-01-27T18:28:36.384+0300 [TRACE] statemgr.Filesystem: creating backup snapshot at terraform.tfstate.d/dev/terraform.tfstate.backup
2022-01-27T18:28:36.387+0300 [TRACE] statemgr.Filesystem: state has changed since last snapshot, so incrementing serial to 141
2022-01-27T18:28:36.387+0300 [TRACE] statemgr.Filesystem: writing snapshot at terraform.tfstate.d/dev/terraform.tfstate
2022-01-27T18:28:36.388+0300 [TRACE] vertex "confluentcloud_kafka_acl.test": visit complete
2022-01-27T18:28:36.388+0300 [TRACE] vertex "provider[\"registry.terraform.io/confluentinc/confluentcloud\"] (close)": starting visit (*terraform.graphNodeCloseProvider)
2022-01-27T18:28:36.389+0300 [TRACE] GRPCProvider: Close
2022-01-27T18:28:36.389+0300 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-01-27T18:28:36.391+0300 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/confluentinc/confluentcloud/0.3.0/linux_amd64/terraform-provider-confluentcloud_0.3.0 pid=128741
2022-01-27T18:28:36.391+0300 [DEBUG] provider: plugin exited
```

How i need to create it in right way?

**Comment by linouk23** [comment]
Interesting, could you run 
```
ccloud kafka acl list --cluster lkc-abc123 --environment env-xyz456
```
to check that ACL indeed wasn't created as a result of running `terraform apply` command?

**Comment by linouk23** [comment]
Oh I see, the issue is you're using 0.3.0 version of TF Provider and it doesn't support "User:sa-abc123" values for a principal attribute yet. You have to manually look up its integer ID: "User:12345", see [this note](https://registry.terraform.io/providers/confluentinc/confluentcloud/latest/docs/guides/sample-project#use-the-confluent-cloud-console-to-inspect-your-resources) for more details.

Luckily, we're expecting to fix this issue in our next 0.4.0 release which is about to happen so it might be a good idea for you to wait for a day or two 🤞 and rerun your config with 0.4.0 version of TF Provider for Confluent Cloud.

**Comment by linouk23** [comment]
👋 @mikhailznak to close the loop, please check out our [0.4.0 release](https://github.com/confluentinc/terraform-provider-confluentcloud/blob/master/CHANGELOG.md#040-january-28-2022) where we fixed the issue!
